### DISCUSSION TITLE
How can I set the default value when using a controller / input mask?

### DISCUSSION THREAD

**Discussion by jvgeee** [react-hook-form/react-hook-form#3625] [Q&A]
Brief is to make a field that looks like `DD/MM/YYYY` and lets the user input a date while retaining the slashes.

I'm using a date field alongside react-input-mask like so:

```
<Controller
      as={<InputMask />}
      control={formControls.control}
      mask="99/99/9999"
      maskPlaceholder="DD/MM/YYYY"
      type={"InputMask"}
      name="dob"
    />
```

Issue: When I do 

```
useForm({defaultValues: {
   dob: "01/01/1910"
}})
```

I get this error: `Warning: A component is changing an uncontrolled input to be controlled. This is likely caused by the value changing from undefined to a defined value, which should not happen. Decide between using a controlled or uncontrolled input element for the lifetime of the component.`

So: how can I set a default value here? Am I perhaps not understanding the use of controllers properly?

**Comment by bluebill1049** [comment]
double-check the example here: https://codesandbox.io/s/react-hook-form-v6-controller-qsd8r in case you missing anything, if still not working send a codesandbox so we can reproduce the issue.

**Comment by bl4ckck** [comment]
@jvgeee Here's an example , wrap the children with a function, let's think of InputMask as Controller and children as render. So I put the ref prop on the children to redirect the ref errors directly to the children or render, not the Controller.

And make sure your data is fetched by using ternary condition to check it

```javascript
{reporterData.npwp ? (
  <InputMask
    name="npwp"
    mask="99.999.999.9-999.999"
    defaultValue={reporterData.npwp} //this is your defaultValue
  >
    {(inputProps) => (
      <input
        {...inputProps}
        ref={register} //your ref
        type="text"
        placeholder="Input NPWP"
      />
    )}
  </InputMask>
) : (
  ""
)}
```

Please correct me if I'm wrong, I'm still learning too :)
